pkg/api/annotations.go

```diff
@@ -211,6 +213,10 @@ func PatchAnnotation(c *models.ReqContext, cmd dtos.PatchAnnotationsCmd) Respons
 		existing.EpochEnd = cmd.TimeEnd
 	}
 
+	if cmd.Data != nil {
```

Comment:
Does one have to pass an empty JSON object in order to erase `Data`?

Reply:
I think it should be a JSON null. I'll add a test for this.

Reply:
Hey @jeet-parekh, do you still hope to add a test for this?

Reply:
If a JSON `null` is passed, won't `cmd.Data` be `nil`? I think in that case I'm right and you have to pass and empty object?

Reply:
@wbrowne, yes. I was waiting for a confirmation on the other thing. So I could push all the necessary changes together.

@aknuds1, you're right. Passing `null/nil` for `cmd.Data` won't do anything. Passing an empty object `{}` will clear the data.